libyul/optimiser/VarNameCleaner.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <libyul/optimiser/VarNameCleaner.h>
+#include <libyul/AsmData.h>
+#include <libyul/Dialect.h>
+#include <algorithm>
+#include <cctype>
+#include <climits>
+#include <iterator>
+#include <string>
+#include <regex>
+
+using namespace yul;
+using namespace std;
+
+void VarNameCleaner::operator()(FunctionDefinition& _funDef)
+{
+	m_usedNames = m_blacklist;
```

Comment:
this is ignored in `operator()(Identifier&` - so I think we should just add that to `m_translatedNames`, too.

Reply:
I think nothing in the blacklist can end up in `m_translatedNames`` and thus the blacklisted names are always left unchanged?

Reply:
Ah right, that's true!